pkg/token/federatedIdentity.go

```diff
@@ -114,3 +111,54 @@ func readJWTFromFS(tokenFilePath string) (string, error) {
 	}
 	return string(token), nil
 }
+
+func getGitHubToken() (string, error) {
+	reqToken := os.Getenv(actionsIDTokenRequestToken)
+	reqURL := os.Getenv(actionsIDTokenRequestURL)
+
+	if reqToken == "" || reqURL == "" {
+		return "", errors.New("ACTIONS_ID_TOKEN_REQUEST_TOKEN or ACTIONS_ID_TOKEN_REQUEST_URL is not set")
+	}
+
+	u, err := url.Parse(reqURL)
+	if err != nil {
+		return "", fmt.Errorf("unable to parse ACTIONS_ID_TOKEN_REQUEST_URL: %w", err)
+	}
+	q := u.Query()
+	q.Set("audience", azureADAudience)
+	u.RawQuery = q.Encode()
+
+	req, err := http.NewRequest("GET", u.String(), nil)
+	if err != nil {
+		return "", err
+	}
+
+	// reference:
+	// https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect
+	req.Header.Set("Authorization", fmt.Sprintf("bearer %s", reqToken))
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Accept", "application/json; api-version=2.0")
+
+	client := http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		body, _ := io.ReadAll(resp.Body)
+		return "", fmt.Errorf("github actions ID token request failed with status code: %d, response body: %s", resp.StatusCode, body)
```

Comment:
the body here might be nil when io.ReadAll failed. Passing a nil value to `%s` is unexpected. And the type of `body` is `[]byte`, I assume you mean to `string(body)` instead? Maybe we can do this:

```
body, readBodyErr := io.ReadAll(resp.Body)
var bodyString string
if readBodyErr == nil {
  bodyString = string(body)
} else {
  bodyString = readBodyErr.Error()
}
```